modules/extension/transform/src/main/java/org/geotools/data/transform/TransformFeatureCollection.java

```diff
@@ -206,4 +208,25 @@ public SimpleFeatureCollection subCollection(Filter filter) {
         }
         return new TransformFeatureCollection(source, transformer, q);
     }
+
+    @Override
+    public void accepts(
+            org.opengis.feature.FeatureVisitor visitor, org.opengis.util.ProgressListener progress)
+            throws IOException {
+        Name typeName = transformer.getSource().getName();
+
+        if (typeName != null) {
+            try {
+                SimpleFeatureSource fs =
+                        (SimpleFeatureSource) source.getDataStore().getFeatureSource(typeName);
+                if (fs instanceof ContentFeatureSource) {
+                    ((ContentFeatureSource) fs).accepts(query, visitor, progress);
```

Comment:
Here is the problem, you cannot, in general, pass down the visitor blindly, but need to check how the property involved was transformed, if at all, and then decide accordingly.
The logic can be easy or complex depending on how far you want to go. My suggestion would be to:
- Recognize CountVisitor and let it go as is
- Recognize FeatureAttributeVisitor implementations, extract the expressions, and see what attributes they are using by using a FilterAttributeExtractor, and then check against the Transformer and see if a rename or a calculation is happening there. 

If the visitors are not a match for the above classes/interfaces, or if any attribute used in the visitor is transformed you cannot pass down the visit and have to rely on super implementation, otherwise you can pass down without issues.